src/AWSCredentials.jl

```diff
@@ -185,263 +190,311 @@ function localhost_is_ec2()
-            creds = aws_get_role(p, ini)
+            return _aws_get_role(p, ini)
         end
     end
 
-    return creds
+    return nothing
 end
 
-function aws_get_role_details(profile::AbstractString, ini::Inifile)
-    if debug_level > 0
-        println("Loading \"$profile\" Profile from " *
-                dot_aws_config_file() * "... ")
-    end
-
-    role_arn = get(ini, profile, "role_arn")
-    source_profile = get(ini, profile, "source_profile")
-
-    profile = "profile $profile"
-    role_arn = get(ini, profile, "role_arn", role_arn)
-    source_profile = get(ini, profile, "source_profile", source_profile)
 
-    (source_profile, role_arn)
-end
+"""
+    _aws_get_credential_details(profile::AbstractString, ini::Inifile, config::Bool)
 
-function aws_get_credential_details(profile::AbstractString, ini::Inifile, config::Bool)
-    if debug_level > 0
-        filename = config ? dot_aws_config_file() : dot_aws_credentials_file()
-        println("Loading \"$profile\" AWSCredentials from " * filename
-                * "... ")
-    end
+# Arguments
+- `profile::AbstractString`: Specific profile used to get AWSCredentials
+- `ini::Inifile`: Inifile to look into for the `profile` credentials
+- `config::Bool`: True if using `~/.aws/config`
 
-    key_id = get(ini, profile, "aws_access_key_id")
-    key = get(ini, profile, "aws_secret_access_key")
+Get `AWSCredentials` for the specified `profile` from the `inifile`. If targeting the `~/.aws/config`
+file, with a non-default `profile`, you must specify `config=true` otherwise the default credentials
+will be returned.
+"""
+function _aws_get_credential_details(profile::AbstractString, ini::Inifile, config::Bool)
+    access_key = get(ini, profile, "aws_access_key_id")
+    secret_key = get(ini, profile, "aws_secret_access_key")
     token = get(ini, profile, "aws_session_token", "")
 
     if config
         profile = "profile $profile"
-        key_id = get(ini, profile, "aws_access_key_id", key_id)
-        key = get(ini, profile, "aws_secret_access_key", key)
+        access_key = get(ini, profile, "aws_access_key_id", access_key)
+        secret_key = get(ini, profile, "aws_secret_access_key", secret_key)
         token = get(ini, profile, "aws_session_token", token)
     end
 
-    (key, key_id, token)
+    return (access_key, secret_key, token)
 end
 
-function aws_get_profile()
-    get(ENV, "AWS_DEFAULT_PROFILE", get(ENV, "AWS_PROFILE", "default"))
-end
 
+"""
+    aws_get_region(profile::AbstractString, ini::Inifile)
+
+# Arguments
+- `profile::AbstractString`: Specific profile used to get the region
+- `ini::Inifile`: Inifile to look in for the region
+
+Retrieve the AWS Region for a given profile, returns `us-east-1` as a default.
+"""
 function aws_get_region(profile::AbstractString, ini::Inifile)
     region = get(ENV, "AWS_DEFAULT_REGION", "us-east-1")
-
     region = get(ini, profile, "region", region)
     region = get(ini, "profile $profile", "region", region)
```

Comment:
I'll just mention this logic is from lowest to highest priority which seems backwards. You could flip things around with do-blocks:
```julia
get(ini, "profile $profile", "region") do
    get(ini, profile, "region") do
        get(ENV, "AWS_DEFAULT_REGION", "us-east-1")
    end
end
```
I'm not convinced my suggestion is actually better or more performant